src/aihwkit/simulator/tiles/module.py

```diff
@@ -227,23 +224,31 @@ def to(self, *args: Any, **kwargs: Any) -> "TileModule":
         if rpu_config is not None:
             self.replace_with(rpu_config)
 
-        device = None
-        if "device" in kwargs:
-            device = kwargs["device"]
-        elif len(new_args) > 0 and not isinstance(new_args[0], (Tensor, dtype)):
-            if new_args[0] is None:
-                return self
-            device = torch_device(args[0])
+        device = kwargs.get("device")
+        dtype = kwargs.get("dtype")
+        for arg in args:
+            if isinstance(arg, bool):
+                continue
+            if isinstance(arg, torch_dtype):
+                dtype = arg
+            if isinstance(arg, (str, torch_device)):
+                device = torch_device(arg) if isinstance(arg, str) else arg
 
         if device is not None:
-            device = torch_device(device)
             if device.type == "cuda":
                 self.cuda(device)
             else:
                 self.cpu()
 
         if len(new_args) > 0 or len(kwargs) > 0:
             self._apply_without_context(lambda t: t.to(*new_args, **kwargs))
+
+        if dtype is not None:
+            # for the torch tile. This renews the hook with the correct dtype
```

Comment:
Should also be the case for the other tiles right? Best delete comment here. Don't specialize generic function for a particular tile.

Reply:
Removed that.